2020/09/02/rclone-upload-to-cloud-from-cli/rclone-upload-to-cloud-from-cli.html.md

```diff
@@ -0,0 +1,239 @@
+ 3 | These are visible in the drive website.
+   | File authorization is revoked when the user deauthorizes the app.
+   \ "drive.file"
+   / Allows read and write access to the Application Data folder.
+ 4 | This is not visible in the drive website.
+   \ "drive.appfolder"
+   / Allows read-only access to file metadata but
+ 5 | does not allow any access to read or download file content.
+   \ "drive.metadata.readonly"
+scope> 3
+```
+
+At this point, choose a scope for the drive access. I'm not sure what the use cases are for the higher numbers, but 1-3 seem to be the most useful. For this demonstration, I chose 3- I want the app to be able to create new files but not read existing ones. For backing up your drive, I recommend 1 or 2.
+
+
+```bash
+ID of the root folder
+Leave blank normally.
+
+Fill in to access "Computers" folders (see docs), or for rclone to use
+a non root folder as its starting point.
+
+Note that if this is blank, the first time rclone runs it will fill it
+in with the ID of the root folder.
+
+Enter a string value. Press Enter for the default ("").
+root_folder_id>
+```
+
+You can also jail the app to a specific subfolder. You'll need to use a folder ID for this, rather than a folder name. (See [the documentation](https://rclone.org/drive/#root-folder-id) if you need to do this)
+
+```bash
+Needed only if you want use SA instead of interactive login.
+Enter a string value. Press Enter for the default ("").
+service_account_file>  #I left this blank
+Edit advanced config? (y/n)
+y) Yes
+n) No (default)
+y/n> n
+Remote config
+Use auto config?
+ * Say Y if not sure
+ * Say N if you are working on a remote or headless machine
+y) Yes (default)
+n) No
+```
+
+For our use case, we are setting this up on a remote machine- but you can use the interactive login if you have a graphical user interface on the machine in question
+
+```bash
+y/n> n
+Please go to the following link: https://accounts.google.com/o/oauth2/auth?access_type=offline&client_id=[snip]
+Log in and authorize rclone for access
+Enter verification code>
+```
+
+At this point copy the URL and follow the instructions on screen. Agree to the data sharing- If you have created your own API key, this data will be only shared with yourself, another reason to create your own.
+
+```bash
+Configure this as a team drive?
+y) Yes
+n) No (default)
+y/n> n
+--------------------
+[gdrive]
+type = drive
+client_id = [snip]
+client_secret = [snip]
+scope = drive.file
+root_folder_id = rclone
+token = [snip]
+--------------------
+y) Yes this is OK (default)
+e) Edit this remote
+d) Delete this remote
+y/e/d> y
+Current remotes:
+
+Name                 Type
+====                 ====
+gdrive               drive
+```
+
+At this point, you can quit the configuration mode, and your new rclone remote is ready to use!
+
+### Mounting the remote drive (Putting it all together)
+
+
+```bash
+$ sudo mkdir /mnt/rclone/
+$ sudo mkdir /mnt/rclone/drive
```

Comment:
For brevity, how about replacing these 2 lines with:

```
sudo mkdir -p /mnt/rclone/drive
```

?

Reply:
today I learned. ;)